src/main/java/build/buildfarm/common/DigestUtil.java

```diff
@@ -62,13 +83,20 @@ public DigestFunction.Value getDigestFunction() {
       if (this == MD5) {
         return DigestFunction.Value.MD5;
       }
+      if (this == BLAKE3) {
+        return DigestFunction.Value.BLAKE3;
+      }
+
       return DigestFunction.Value.UNKNOWN;
     }
 
     public static HashFunction forHash(String hexDigest) {
       if (SHA256.isValidHexDigest(hexDigest)) {
         return SHA256;
       }
+      if (BLAKE3.isValidHexDigest(hexDigest)) {
+        return BLAKE3;
+      }
```

Comment:
BLAKE3 and SHA256 both have 256 bit digest, With this logic, we can't identify the hashfn by digest.